### PR TITLE
[RLlib] Documentation do-over 01: Announce new API stack as alpha; add hints to all RLlib pages; describe how to use it in new page.

### PR DESCRIPTION
**Pull request by sven1977**
Documentation do-over 01:
* Announce new API stack as alpha by adding `Note:`s on top of all RLlib pages
* Add documentation on how to switch to the new stack (in a new page). The before-mentioned notes link to this new page. Furthermore, from this page, users can get to the already existing API documentation pages for the new API components, such as RLModule and Learner.

TODO (in follow up PRs):
* Describe ConnectorV2 APIs.
* Describe new Episode APIs.
* Describe EnvRunner APIs.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by zhe-thoughts — APPROVED** [review]
This is a large PR, but docs only changes

File: doc/source/rllib/rllib-new-api-stack.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+1) You are using custom :py:class:`~ray.rllib.connector.connector.Connector` classes and are not interested right now in moving them into the new :py:class:`~ray.rllib.connector.connector_v2.ConnectorV2` API.
+
+If any of the above applies to you, simply stay put for now, continue running with the old API stack, and cut over to the new
+stack whenever you feel ready to re-write some (small) part of your code.
+
+
+Comparison to the Old API Stack
+-------------------------------
+
+Here is a quick comparison table listing features and design choices from the new- vs the old API stack:
```
Comment (angelinalg) [inline doc/source/rllib/rllib-new-api-stack.rst:103]:
```suggestion
This table compares features and design choices between the new and old API stack:
```

File: doc/source/rllib/rllib-new-api-stack.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+   * -
+     - **New API Stack**
+     - **Old API Stack**
+   * - Reduced code complexity (for beginners and advanced users)
+     - 5 user-facing classes (`AlgorithmConfig`, `RLModule`, `Learner`, `ConnectorV2`, `Episode`)
+     - 8 user-facing classes (`AlgorithmConfig`, `ModelV2`, `Policy`, `build_policy`, `Connector`, `RolloutWorker`, `BaseEnv`, `ViewRequirement`)
+   * - Classes are usable outside of RLlib
+     - Yes
+     - Partly
+   * - Separation-of-concerns design (e.g. during sampling, only action must be computed)
```
Comment (angelinalg) [inline doc/source/rllib/rllib-new-api-stack.rst:118]:
```suggestion
   * - Separation-of-concerns design (e.g., during sampling, only action must be computed)
```

File: doc/source/rllib/rllib-new-api-stack.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+   * - Distributed/scalable sample collection
+     - Yes
+     - Yes
+   * - Full 360° read/write access to (multi-)agent trajectories
+     - Yes
+     - No
+   * - Multi-GPU & multi-node/multi-GPU
+     - Yes
+     - Yes & No
+   * - Support for shared (multi-agent) model components (e.g. communication channels, shared value functions, etc..)
```
Comment (angelinalg) [inline doc/source/rllib/rllib-new-api-stack.rst:130]:
```suggestion
   * - Support for shared (multi-agent) model components (e.g., communication channels, shared value functions, etc.)
```

File: doc/source/rllib/rllib-new-api-stack.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+     - Yes
+     - No
+   * - Env vectorization via `gym.vector.Env`
```
Comment (angelinalg) [inline doc/source/rllib/rllib-new-api-stack.rst:133]:
```suggestion
   * - Env vectorization with `gym.vector.Env`
```

**Review by angelinalg — APPROVED** [review]

**Comment by angelinalg** [comment]
Just had some nits

**Pull request merged**